tests/e2e/e2e_retry_policy_test.go

```diff
@@ -0,0 +1,291 @@
+package e2e
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openservicemesh/osm/pkg/apis/policy/v1alpha1"
+	. "github.com/openservicemesh/osm/tests/framework"
+)
+
+const client = "client"
+const server = "server"
+
+var meshNs = []string{client, server}
+
+var retryStats = map[string]string{"upstream_rq_retry": "", "upstream_rq_retry_limit_exceeded": "", "upstream_rq_retry_backoff_exponential": ""}
+var thresholdUintVal uint32 = 5
+
+var _ = OSMDescribe("Test Retry Policy",
+	OSMDescribeInfo{
+		Tier:   2,
+		Bucket: 8,
```

Comment:
Nit: move to bucket 4 which typically completes sooner than other buckets. This helps in keeping the bucket times more balanced.